datafusion/physical-plan/src/joins/sort_merge_join.rs

```diff
@@ -1363,6 +1380,57 @@ fn get_filter_column(
     filter_columns
 }
 
+// Get buffered data sliece by specific batch index and for specified column indices only
+#[inline(always)]
+fn get_buffered_columns(
+    buffered_data: &BufferedData,
+    buffered_batch_idx: usize,
+    buffered_indices: &UInt64Array,
+) -> Result<Vec<ArrayRef>, ArrowError> {
+    buffered_data.batches[buffered_batch_idx]
+        .batch
+        .columns()
+        .iter()
+        .map(|column| take(column, &buffered_indices, None))
+        .collect::<Result<Vec<_>, ArrowError>>()
+}
+
+// Calculate join filter bit mask considering join type specifics
```

Comment:
I think it would help to also document what the input parameters `streamed_indicies` and `mask` represents here (I think it is the rows in streamed_indices that match the join predicate?)

Reply:
Will do as a few documentation also added me quite a pain